refactor(give-block): Update give block loading animation

Sidsector9 opened the pull request:
Closes #2843 

Removed the give logo animation and replaced it with facebook-like placeholder animation.

## Description
This is a further improvisation on @ajitbohra's PR #2799 

This [animation](https://user-images.githubusercontent.com/1039236/36101077-52b428da-102e-11e8-96ba-f28da62b0967.gif) has been replaced. See the screenshot section below for the video which shows the replacement.

## How Has This Been Tested?
Manually tested.

## Screenshots:
Check this [video](https://drive.google.com/file/d/1-Lv-z9LArrXQGBM0Ro61j0_usHxi5ECs/view) for the output

## Merge Checklist:
- [x] Polish loading UI
- [x] Break out SCSS into component based files rather than a global `style.scss`
- [x] Ensure the "G" icon isn't green then but rather the same color as other native WP block icons

mehul0810 commented:
@DevinWalker 
I've checked the video uploaded by @Sidsector9 and i liked the way he has added the loading section for the Give Gutenberg block. However, I would suggest that we can have Give Logo loaded by default and rest of the items to be loading placeholder as shown in the video until the complete block is not loaded.

**For Example**,
1. When a user clicks on Give Gutenberg Block, then loading will have Give Logo and below it, there will be a grey box for donation forms dropdown and below that a grey box for buttons.
2. Now, when a user selects the donation form, then the loading placeholder will have the grey boxes placement similar to the way the donation form view.

**Give Loading Placeholder Image**
![placeholder-give-block1](https://user-images.githubusercontent.com/1852711/36735658-ad629d96-1bfc-11e8-9118-2e1f3dfe3cd1.jpg)

Let me know your thoughts on this.

DevinWalker commented:
@Sidsector9 @mehul0810 I also like the animation. I agree with @mehul0810 that we should add the Give icon in there and also make the fields look more life _form fields_.

Here's a rough concept:

![36735658-ad629d96-1bfc-11e8-9118-2e1f3dfe3cd1](https://user-images.githubusercontent.com/1571635/36756208-3b4f7442-1bc3-11e8-9ce2-3ea592f43186.jpg)

ajitbohra commented:
Super awesome someone reading my mind :) was on my wishlist to add this. 

As mentioned by @mehul0810 it would be good to have Give logo on loading animation & also on form selection.

Saw the video & changes I guess it removes the Logo/Icon from BlankSlate component which is currently used at some place & possibly be used in future.

Have added the suggestions in code review. Might sound bit picky :D just getting older :p  

Feel free to ignore them ;)

Sidsector9 commented on blocks/donation-form/style.scss:
```diff
@@ -5,3 +5,139 @@
 .float-right {
```
@ajitbohra Yes, component-specific CSS is being discussed with @ravinderk already. I'm on with you on this one. For the time being until we figure out how to implement it, we'll be putting all the CSS in a single file.

mathetos commented:
I like the new animation. Well done!

But I don't think we should be adding a Give inserter into the Add New section here:
![image](https://user-images.githubusercontent.com/930724/36812503-3f11279c-1c86-11e8-8494-2db88d722937.png)

It's a similar conversation to when all plugins were trying to add their own unique buttons into the TinyMCE toolbar areas. That area got bloated really quickly. If every addon tried to add their own logo and button there, it'd get totally out of control.

It's early in Gutenberg plugin adoption history right now, so it's hard to say where this will go, but I believe the "+" to insert a block is there specifically to avoid that kind of bloat, and we should be thinking of Gutenberg best practices.

DevinWalker commented:
Since this PR is concerning more visual elements I broke out the list into this PR: https://github.com/WordImpress/Give/pull/2856 which contains the work being done on the feature branch.

ajitbohra commented:
@mathetos @DevinWalker update on "Give inserter into the Add New section here" Gutenberg displays frequently used blocks in inserter bar, it's not forced by the plugin. I am not convinced though i guess last registered blocks, particularly under "common" category, are taking prominence.

https://github.com/WordPress/gutenberg/issues/5312

DevinWalker commented:
Thanks @ajitbohra if that's so then it clears up some confusion. We should really make sure our icon isn't green then but rather the same color as other block icons. I'll add that to the list.

ajitbohra commented:
@DevinWalker moving block to widget category will help us fix it.

ajitbohra commented:
@DevinWalker

"Ensure the "G" icon isn't green then but rather the same color as other native WP block icons"

implemented in #2859

DevinWalker commented:
@Sidsector9 please provide a status on this one. I'd like to get it merged once the animation looks nice.

Sidsector9 commented:
@DevinWalker  PR #2856 is merged into this.

I have added component specific stylesheet.

As for polishing the loading UI, I have implemented it similar to how you asked [here](https://github.com/WordImpress/Give/pull/2844#issuecomment-369034675)
I need your feedback on the animation.

Merge Checklist
- [x] Remove "G" colored icon from block bar #2859
- [ ] Polish loading UI #2844
- [x] Use WP REST API rather than Give's API so users don't have to create an API key #2859
- [x] Break out SCSS into component based files rather than a global style.scss #2844

ravinderk commented on blocks/components/blank-slate/index.js:
```diff
@@ -19,24 +20,34 @@ const GiveBlankSlate = ( props ) => {
 
+
 	// @todo: do not hard code wp content url that can be configure.
 	const giveLoader = '/wp-content/plugins/Give/assets/dist/images/give-loader.svg';
```
@sid remove unused loader and svg file

ravinderk commented:
@Sidsector9 This pr updated with https://github.com/WordImpress/Give/pull/2856

Sidsector9 commented:
@DevinWalker @ravinderk I have changed the look of the control buttons, please have a look here - https://drive.google.com/file/d/1KG-R_2hi8ABtXocHETz1KMT3WQJO4e9X/view

ajitbohra commented:
@Sidsector9 
New controls look good

**Personal Thoughts:**
Would be good to follow Gutenberg UX/UI for interacting with the block.
Currently Gutenberg provides us with the `BlockControls` & 'InspectorControls'.

@DevinWalker @kevinwhoffman your thoughts?

Sidsector9 commented:
@ajitbohra Thank you :smile: 

The new controls were put after discussion with @DevinWalker and @ravinderk to make it look more interactive with animation.

Originally `BlockControls` and `InspectorControls` and I thought of implementing the UI using those, but they came with their own CSS, so had to remove them completely.

ajitbohra commented:
@Sidsector9 cool, new interaction is looking good 👍
